marklogic-client-api/src/main/java/com/marklogic/client/dataservices/impl/OutputEndpointImpl.java

```diff
@@ -79,14 +87,17 @@ public void awaitCompletion() {
                     throw new RuntimeException("error while calling "+getEndpoint().getEndpointPath(), throwable);
                 }
 
+                InputStream[] result = null;
```

Comment:
Should the result variable and all of the result processing move inside the branch that tests output is not null?  If there's no output, by definition, there's no result processing.